### PR TITLE
enh(conf/export) list associated severity/tags in host/host templates/services templates export

### PR DESCRIPTION
**Pull request by a-launois**
## Description

export tags and severity in host, host templates, and service templates config files

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x (master)

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/class/config-generate/servicecategory.class.php [hunk 1]
```diff
@@ -168,4 +168,16 @@ public function reset(): void
             }
         }
     }
+
+    public function getServiceCategoryIdsForService(int $serviceId): array
```
Comment (kduret) [inline www/class/config-generate/servicecategory.class.php:172]: missing phpdoc
could be renamed to `getIdsByServiceId`

File: www/class/config-generate/servicecategory.class.php [hunk 2]
```diff
@@ -168,4 +168,16 @@ public function reset(): void
...
+    {
+        $serviceCategoryIds = [];
+        foreach ($this->serviceCategories as $id => &$value) {
```
Comment (kduret) [inline www/class/config-generate/servicecategory.class.php:175]:
```suggestion
        foreach ($this->serviceCategories as $id => $value) {
```

File: www/class/config-generate/hostcategory.class.php [hunk 3]
```diff
@@ -129,4 +129,20 @@ public function reset(): void
...
     }
+
+    /**
+     * @param int $hostId
+     * @return int[]
+     */
+    public function getHostCategoryIdsForHost(int $hostId): array
+    {
+        $hostCategoryIds = [];
+        foreach ($this->hostCategories as $id => &$value) {
```
Comment (kduret) [inline www/class/config-generate/hostcategory.class.php:140]:
```suggestion
        foreach ($this->hostCategories as $id => $value) {
```

File: www/class/config-generate/hostcategory.class.php [hunk 4]
```diff
@@ -129,4 +129,20 @@ public function reset(): void
             }
         }
     }
+
+    /**
+     * @param int $hostId
+     * @return int[]
+     */
+    public function getHostCategoryIdsForHost(int $hostId): array
```
Comment (kduret) [inline www/class/config-generate/hostcategory.class.php:137]: could be renamed to `getIdsByHostId`

File: www/class/config-generate/host.class.php [hunk 5]
```diff
@@ -404,9 +404,16 @@ protected function getSeverity($host_id_arg)
         $severity_id = $severity_instance->getHostSeverityByHostId($host_id_arg);
         $this->hosts[$host_id_arg]['severity'] = $severity_instance->getHostSeverityById($severity_id);
         if (!is_null($this->hosts[$host_id_arg]['severity'])) {
-            $this->hosts[$host_id_arg]['macros']['_CRITICALITY_LEVEL'] =
-                $this->hosts[$host_id_arg]['severity']['level'];
-            $this->hosts[$host_id_arg]['macros']['_CRITICALITY_ID'] = $this->hosts[$host_id_arg]['severity']['hc_id'];
+            $severityMacros = [
+                '_CRITICALITY_LEVEL' => $this->hosts[$host_id_arg]['severity']['level'],
+                '_CRITICALITY_ID' => $this->hosts[$host_id_arg]['severity']['hc_id'],
+                'severity' => $this->hosts[$host_id_arg]['severity']['hc_id'],
```
Comment (kduret) [inline www/class/config-generate/host.class.php:410]: why `severity` is a macro ?

File: www/class/config-generate/servicecategory.class.php [hunk 6]
```diff
@@ -168,4 +168,16 @@ public function reset(): void
             }
         }
     }
+
+    public function getIdsByServiceId(int $serviceId): array
```
Comment (a-launois) [inline www/class/config-generate/servicecategory.class.php:172]:
```suggestion
 
    /**
     * @param int $serviceId
     * @return int[]
     */
    public function getIdsByServiceId(int $serviceId): array
```

**Review by kduret — APPROVED** [review]

File: www/class/config-generate/servicecategory.class.php [hunk 7]
```diff
@@ -168,4 +168,20 @@ public function reset(): void
             }
         }
     }
+
```
Comment (a-launois) [inline www/class/config-generate/servicecategory.class.php:171]:
```suggestion

```

**Review by callapa — APPROVED** [review]

**Pull request merged**